Initial deb and rpm packaging

jchengsfx opened the pull request:
**Description:**
- Uses https://github.com/jordansissel/fpm
- Adds `deb-package` and `rpm-package` targets to Makefile
- Builds simple deb and rpm packages that installs the otelcontribcol
binary to `/usr/bin/otelcontribcol`
- Adds deb and rpm packages to circleci workspace for github release
- Installs `otel-contrib-collector` systemd service and run as `otel:otel` user:group

**Testing:**
- Added simple installation test after package build using systemd-in-docker images

**Documentation:**
- Added `deb/README.md` and `rpm/README.md`

jchengsfx committed:
Initial deb and rpm packaging

- Uses https://github.com/jordansissel/fpm
- Adds `deb-package` and `rpm-package` targets to Makefile
- Builds simple deb and rpm packages that installs the otelcontribcol
binary to `/usr/bin/otelcontribcol`
- Adds deb and rpm packages to circleci workspace for github release
- TODO: create service scripts

owais commented on .circleci/config.yml:
```diff
@@ -351,3 +367,34 @@ jobs:
+          command: |
+            sudo apt-get update
+            sudo apt-get install -y ruby ruby-dev rubygems build-essential rpm
+            sudo gem install --no-document fpm -v 1.11.0
```
Instead of all this, can this job use docker executor with an FPM image (such as `alanfranz/fpm-within-docker`) ?

jchengsfx replied:
I looked but couldn't find an existing image that pinned the fpm version and had the ability to build both deb and rpm packages.

owais replied:
Looks like the one I mentioned does support both deb and rpm. It ues 1.11 right now but probably will upgrade without adding a new tag. I see we have custom images for local builds, may be we can add support for both deb and rpm to the local dockerfiles and publish to otel org, and then use that in CI (I can help publish)

jchengsfx replied:
The local build has been updated to use a single image for both rpm and deb.

I think using a published image for CI will eventually be the way to go.  But for now, since the packaging is still under development and may require changes to the Dockerfile and then re-publishing the image, we can avoid these extra steps with just doing the setup in the circleci job.  Thoughts?

owais replied:
Sounds good.

owais reviewed (commented):
Don't we also want to publish deb/rpm packages for ARM?

owais commented on packaging/deb/build.sh:
```diff
@@ -0,0 +1,23 @@
+    --url "https://github.com/open-telemetry/opentelemetry-collector-contrib" \
+    --architecture "x86_64" \
+    --deb-dist "stable" \
+    $OTELCONTRIBCOL_PATH=/usr/bin/otelcontribcol
```
I suppose we store the packages in `bin/` so CI release job automatically uploads the packages to github releases? Can we add that as a comment so someone in future doesn't move them away from `bin/` to something like `packaging/dist` and break the release script?

jchengsfx replied:
I updated the circleci job to write the packages to the `bin/` directory, so someone would have to change the circleci config to move the packages to a different directory.

jchengsfx commented:
> Don't we also want to publish deb/rpm packages for ARM?

Good point. Will update to build ARM packages as well.

jchengsfx committed:
Packaging updates

- Add support for arm64 packages
- Move files to internal/buildscripts/packaging/
- Use single Dockerfile for local deb and rpm builds
- Explicitly set output dir for packages in circleci

jchengsfx committed:
Add systemd service to deb and rpm

- Install `otel-contrib-collector` systemd service and run as `otel:otel` user:group
- Add simple installation test scripts using systemd-in-docker images

owais committed:
wip

owais commented:
@jchengsfx can you update this PR to reflect the changes introduced in https://github.com/open-telemetry/opentelemetry-collector/pull/1360? Let's merge this today before we release v0.6

jchengsfx committed:
Change default output dir to dist

jchengsfx committed:
Fix publish CI job and add check to make sure publish doesn't break
